### PR TITLE
Update to prettier 3

### PR DESCRIPTION
**Pull request by clementdessoude**
## What changed with this PR:

Update prettier dependency to v3

## Relative issues or prs:

#557

### CODE REVIEW

**Comment by fisker** [comment]
You may want remove `@types/prettier`

**Comment by clementdessoude** [comment]
> You may want remove `@types/prettier`

Yes, indeed ! Thanks for pointing this out @fisker ! I think everything is working fine with prettier v3 here.

I will keep this issue opened until the final prettier v3.0.0 is released, and update it with intermediate version if there are any

File: packages/prettier-plugin-java/src/printers/prettier-builder.ts [hunk 1]
```diff
@@ -2,8 +2,6 @@
 import { IToken } from "java-parser";
 import { doc } from "prettier";
 import Doc = doc.builders.Doc;
-import Concat = doc.builders.Concat;
-import Fill = doc.builders.Fill;
 
 const prettier = require("prettier").doc.builders;
```
Comment (fisker) [inline packages/prettier-plugin-java/src/printers/prettier-builder.ts:6]: I think this  require() can be removed too. `doc` is imported, and it's not type only.

**Comment by clementdessoude** [comment]
Oh, yeah! 🥳
I'll try to update this weekend!

**Pull request merged**